### PR TITLE
tests: make the TSO testing as independent as possible

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #6181.

### What is changed and how does it work?

```commit-message
- Support testing both legacy and microservice TSO.
- Make the TSO testing as independent as possible.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by JmPotato** [comment]
/cc @binshi-bing @rleungx @lhy1024 

PTAL

File: client/tso_client.go [hunk 1]
```diff
@@ -29,11 +29,16 @@ import (
 	healthpb "google.golang.org/grpc/health/grpc_health_v1"
 )
 
-// TSOClient defines basic interface of the TSO client
-// For test only
+// TSOClient is the client used to get TSO.
```
Comment (binshi-bing) [inline client/tso_client.go:32]:
```suggestion
// TSOClient is the client used to get timestamps.
```

File: tests/client/client_test.go [hunk 2]
```diff
@@ -512,7 +472,10 @@ func requestGlobalAndLocalTSO(
 	wg.Wait()
 }
 
-func getTSOAllocatorServingEndpointURLs(c pd.TSOClient) map[string]string {
+// GetTSOAllocators defines the TSO allocators getter.
+type TSOAllocatorsGetter interface{ GetTSOAllocators() *sync.Map }
```
Comment (binshi-bing) [inline tests/client/client_test.go:476]: +1

File: tests/tso/client_test.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+		legacy: false,
+	})
+}
+
+func (suite *tsoClientTestSuite) SetupSuite() {
+	re := suite.Require()
+
+	var err error
+	suite.ctx, suite.cancel = context.WithCancel(context.Background())
+	switch suite.legacy {
```
Comment (binshi-bing) [inline tests/tso/client_test.go:66]: interesting. We use switch for bool. I'm ok with it which seems to be personal preference.

File: tests/tso/client_test.go [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+	case true:
+		suite.cluster, err = tests.NewTestCluster(suite.ctx, serverCount)
```
Comment (binshi-bing) [inline tests/tso/client_test.go:68]: Line 68 ~ 74 seems can be shared by the two cases.

**Review by binshi-bing — APPROVED** [review]
LGTM

File: tests/tso/go.mod [hunk 5]
```diff
@@ -0,0 +1,173 @@
+module github.com/JmPotato/pd/tests/tso
```
Comment (rleungx) [inline tests/tso/go.mod:1]: tikv?

File: tests/tso/client_test.go [hunk 6]
```diff
@@ -0,0 +1,173 @@
+// Copyright 2023 TiKV Project Authors.
```
Comment (rleungx) [inline tests/tso/client_test.go:1]: BTW, do we need a Makefile since we use it to filter the sub-module?

**Review by rleungx — APPROVED** [review]

**Review by binshi-bing — APPROVED** [review]
first LGTM after I became the reviewer

**Pull request merged**